### ISSUE TITLE
Algolia search ignores URL hash

### ISSUE THREAD

**Issue by wesgro** [vuejs/vuepress#1775]
- [ x] I confirm that this is an issue rather than a question.

#### Steps to reproduce

* Have Algolia search setup
* Navigate to a page that has a secondary heading so there is a page anchor to navigate to
* When you switch to the new page you're sent to the top of the page, not to the anchor.

#### What is expected?
* The page jumps to the anchor

#### What is actually happening?
* The page does not jump to the anchor, it stays at the top of the page

#### Other relevant information

- Output of `npx vuepress info` in my VuePress project:
```
Environment Info:

  System:
    OS: macOS 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
  Binaries:
    Node: 11.15.0 - ~/.nvm/versions/node/v11.15.0/bin/node
    Yarn: 1.17.3 - ~/.yarn/bin/yarn
    npm: 6.7.0 - ~/.nvm/versions/node/v11.15.0/bin/npm
  Browsers:
    Chrome: 76.0.3809.100
    Firefox: 65.0.1
    Safari: 12.1.2
  npmPackages:
    @vuepress/core:  1.0.3 
    @vuepress/theme-default:  1.0.3 
    vuepress: ^1.0.0 => 1.0.3 
  npmGlobalPackages:
    vuepress: Not Found
```

**Comment by wesgro** [comment]
I believe this is from https://github.com/vuejs/vuepress/commit/644142b

Which leaves out the `hash` from the url

**Comment by sarahdayan** [comment]
Fixed in #1780
